src/hooks/useGlobalAppState.ts

```diff
@@ -1,116 +1,51 @@
-import Minilog from 'cozy-minilog'
-import { NavigationContainerRef } from '@react-navigation/native'
-import { Route } from '@react-navigation/routers'
-import { useEffect } from 'react'
-import {
-  AppState,
-  AppStateStatus,
-  NativeEventSubscription,
-  Platform
-} from 'react-native'
-
-import * as RootNavigation from '/libs/RootNavigation'
-import { getData, StorageKeys, storeData } from '/libs/localStore/storage'
-import { routes } from '/constants/routes'
-import {
-  showSplashScreen,
-  hideSplashScreen
-} from '/app/theme/SplashScreenService'
-import { determineSecurityFlow } from '/app/domain/authorization/services/SecurityService'
+import { useNavigationState } from '@react-navigation/native'
+import { useEffect, useRef } from 'react'
+import { AppState, AppStateStatus, NativeEventSubscription } from 'react-native'
 
+import Minilog from 'cozy-minilog'
 import CozyClient, { useClient } from 'cozy-client'
 
+import { StorageKeys, storeData } from '/libs/localStore/storage'
+import { showSplashScreen } from '/app/theme/SplashScreenService'
+import {
+  getIsSecurityFlowPassed,
+  handleSecurityFlowWakeUp,
+  setIsSecurityFlowPassed
+} from '/app/domain/authorization/services/SecurityService'
 import { devlog } from '/core/tools/env'
 import { synchronizeDevice } from '/app/domain/authentication/services/SynchronizeService'
 import { safePromise } from '/utils/safePromise'
+import {
+  initSharingMode,
+  isSharingMode,
+  setFilesToUpload
+} from '/app/domain/sharing/SharingService'
+import { routes } from '/constants/routes'
+import { navigate } from '/libs/RootNavigation'
 
 const log = Minilog('useGlobalAppState')
 
-const TIMEOUT_VALUE = 5 * 60 * 1000
-
 // Runtime variables
 let appState: AppStateStatus = AppState.currentState
-
-const tryLockingApp = async (
-  parsedRoute: Route<string, { href: string; slug: string }>,
-  client: CozyClient
-): Promise<void> => {
-  devlog('tryLockingApp with', { parsedRoute, client })
-
-  return await determineSecurityFlow(
-    client,
-    // Let's assume the parsedRoute could be undefined for unknown reasons
-    // In that case we just don't pass the navigation object and the security flow will default to home
-    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-    parsedRoute?.name !== 'default'
-      ? {
-          navigation: RootNavigation.navigationRef as NavigationContainerRef<
-            Record<string, unknown>
-          >,
-          href: parsedRoute.params.href,
-          slug: parsedRoute.params.slug
-        }
-      : undefined
-  )
-}
+initSharingMode()
```

Comment:
If this is called in the script's root, this means we have no control of when this is called (i.e. this will be called as soon as we import `useGlobalAppState.ts`). So why putting this here but not directly in the `SharingService.ts`? (although it would be greater if done somewhere in the react lifecycle)

Reply:
yes initially I put it in App.js. I think I should use it once it the main useGlobalAppState() hook (which is rendered by App).